gwdatafind/utils.py

```diff
@@ -137,6 +140,66 @@ def find_credential():
                        "Please run 'grid-proxy-init -rfc' and try again.")
 
 
+def find_scitoken(aud, scope):
+    """Locate a Scitoken for a given audience and scope.
+
+    This function checks for a SciToken using the WLCG token discovery
+    protocol. If token is found that matches the requested audience and scope
+    then it is returned. If no matching token is found using this protocol,
+    the function checks to see if HTCondor has created a SciToken with the
+    appropriate audience and scope in the _CONDOR_CREDS directory. Tokens in
+    _CONDOR_CREDS that do not have the correct audience and scope are ignored.
+
+    Parameters
+    ----------
+    aud : `str`
+        the audience of the scitoken
+
+    scope : `str`
+        the scope of the scitoken
+
+    Returns
+    -------
+    token_data : `SciToken`
+        the located token or None if no token for the specified
+        audience can be found
+
+    """
+    scheme, path = scope.split(':')
+    token = None
+
+    # look for a token in with WLCG Bearer Token Discovery protocol
+    try:
+        token = scitokens.SciToken.discover(audience=aud)
+        token_enforcer = scitokens.Enforcer(token['iss'], audience=aud)
+        if token_enforcer.test(token, scheme, path) is False:
+            raise ValueError
+    except:
+        # look for a condor-created scitoken for the target url
+        if os.environ.get('_CONDOR_CREDS'):
+            for f in os.listdir(os.environ['_CONDOR_CREDS']):
+                if f.endswith(".use"):
+                    try:
+                        with open(os.path.join(
+                                  os.environ['_CONDOR_CREDS'],f)) as t:
+                            token_data = t.read().strip()
+                            token = scitokens.SciToken.deserialize(token_data,
+                                                                 audience=aud)
+                            token_enforcer = scitokens.Enforcer(token['iss'],
+                                                                audience=aud)
+                            if token_enforcer.test(token, scheme, path) is False:
+                                raise ValueError
```

Comment:
This block is identical to the block from L174 above, so should probably be moved into a function and called in both places.